src/plugins/vis_builder/public/application/utils/state_management/metadata_slice.ts

```diff
@@ -51,9 +51,9 @@ export const slice = createSlice({
   name: 'metadata',
   initialState,
   reducers: {
-    setValidity: (state, action: PayloadAction<{ key: string; valid: boolean }>) => {
-      const { key, valid } = action.payload;
-      state.editor.validity[key] = valid;
+    setError: (state, action: PayloadAction<{ key: string; error: boolean }>) => {
```

Comment:
You are right, this was one of my not so well thought out implementations to track error states across the app. The original intention of this key was to let the app know that an error has occurred somewhere in the app and using the error key we can identify where the error occurred. The responsibility of displaying and handling the error remains with the component that saw the error. This was a quick and dirty way of letting for example the Save button know that it should be disabled because the config panel values are incorrect. This may see a big redesign in future, I just focussed on renaming it in this PR because `errors` was more intuitive than `validity`